lib/psulib_traject/solr_manager.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'config'
+require 'faraday'
+
+module PsulibTraject
+  # For interacting with Apache Solr, an incomplete implementation of
+  # https://github.com/psu-libraries/psulib_blacklight/blob/master/lib/psulib_blacklight/solr_manager.rb
+  class SolrManager
+    COLLECTION_PATH = '/solr/admin/collections'
+
+    def initialize
+      Config.setup do |config|
+        config.const_name = 'ConfigSettings'
+        config.use_env = true
+        config.load_and_set_settings(Config.setting_files('config', ENV['RUBY_ENVIRONMENT']))
+      end
+    end
+
+    def last_incremented_collection
+      collections_with_prefix.max_by(&:version_number)
+    end
+
+    private
+
+    def collections_with_prefix
+      collections.select { |c| c.name.scan(/#{ConfigSettings.solr.collection_name}/) }
+    end
+
+    def collections
+      resp = connection.get(COLLECTION_PATH, action: 'LIST')
+      JSON.parse(resp.body)['collections'].map { |collection| SolrCollection.new(collection) }
+    end
+
+    def connection
+      @connection ||= Faraday.new(ConfigSettings.solr.url) do |faraday|
+        faraday.request :multipart
+        faraday.adapter :net_http
+      end
+    end
+  end
+
+  # Lightweight Solr collection abstraction
+  class SolrCollection
```

Comment:
nice 👍 maybe we implement this abstraction to BlackCat solrmanager?

Reply:
Yeah, I thought this was an improvement too, wanted to hear your thoughts. I think I wish SolrManager was just a separate plugin we could use anywhere, if time were infinite that's what I'd prefer

Reply:
yes! how nice that would be 💯